index.js

```diff
@@ -12,11 +14,34 @@ exports.lintText = (str, opts) => {
 		delete opts.overrides;
 
 		const filename = path.relative(opts.cwd, opts.filename);
+
 		const foundOverrides = optionsManager.findApplicableOverrides(filename, overrides);
 		opts = optionsManager.mergeApplicableOverrides(opts, foundOverrides.applicable);
 	}
 
 	opts = optionsManager.buildConfig(opts);
+	const defaultIgnores = optionsManager.getIgnores({}).ignores;
+
+	if (opts.ignores && !arrayEqual(defaultIgnores, opts.ignores) && typeof opts.filename !== 'string') {
```

Comment:
I don't really understand the purpose of the `!arrayEqual(defaultIgnores, opts.ignores)` check. ?

I would also prefer using `lodash.isequal` instead as it's more general and could be used for other deep checks.

It should also check that the `cwd` option is defined.